pallets/xtransfer/src/xcm/xcm_transfer.rs

```diff
@@ -295,27 +321,42 @@ pub mod pallet {
 			}
 			.into();
 
+			// if self.asset.id == self.fee.id, self.asset must contains self.fee
+			let withdraw_asset = if self.asset.contains(&self.fee) {
+				// merge
+				(WithdrawAsset(self.asset.clone().into()), 1)
+			} else {
+				(
+					WithdrawAsset(vec![self.asset.clone(), self.fee.clone()].into()),
+					2,
+				)
+			};
+
 			let deposit_asset = DepositAsset {
 				assets: Wild(All),
-				max_assets: 1u32,
+				max_assets: withdraw_asset.1,
 				beneficiary: beneficiary.into(),
 			};
 
 			let kind = self.kind().ok_or(Error::<T>::UnknownTransfer)?;
 			log::trace!(target: LOG_TARGET, "Transfer type is {:?}.", kind.clone(),);
 			let message = match kind {
-				TransferType::FromNative => Xcm(vec![TransferReserveAsset {
-					assets: self.asset.clone().into(),
-					dest: self.dest_location.clone(),
-					xcm: Xcm(vec![
-						self.buy_execution_on(&self.dest_location)?,
-						deposit_asset,
-					]),
-				}]),
+				TransferType::FromNative => Xcm(vec![
+					withdraw_asset.0,
+					DepositReserveAsset {
+						assets: Wild(All),
+						max_assets: withdraw_asset.1,
+						dest: self.dest_location.clone(),
+						xcm: Xcm(vec![
+							self.buy_execution_on(&self.dest_location)?,
```

Comment:
Oh, it turns out `buy_execution_on` does the reanchor. Can we add a comment in its docstring?

Reply:
Comments added.